### PR TITLE
VFB-42 Clean Main

### PR DESCRIPTION
**Pull request by SamYap-Softwire**
This PR cleans minor inconsistencies and issues in main but doesn't do any major refactoring like grouping functions and files. 

1. Change all types to pascal case
2. Add "use client" to all files that use styled components or react hooks 
3. Change type of all `React.FC` children to `React.ReactNode`
4. Change all imports to be absolute import path (@)
5. Add ES Lint Rules for imports and exports 
6. Check all `supabase` imports are correct (client and server)
7. Rename default exports to be consistent with filename where possible 

- [x] Make sure you've verified it works on dev via `npm run dev`
- [x] Make sure you've verified it works on prod via `npm run build` and `npm run start`
- [x] Make sure you've fixed all linting problems with `npm run lint_fix`
  - Will also be automatically checked on pull request!
- [x] Make sure you've tested via `npm run test`
  - Will also be automatically checked on pull request!

### CODE REVIEW

File: src/app/calendar/getParcelsWithCollectionDate.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import supabase from "@/supabaseServer";
+import { Schema } from "@/database_utils";
+
+export type ParcelWithClientName = Schema["parcels"] & { clients: { full_name: string } | null };
+
+export interface LocationColorMap {
```
Comment (SamYap-Softwire) [inline src/app/calendar/getParcelsWithCollectionDate.ts:6]: not used - delete

File: src/app/calendar/parcelsToCollectionEvents.ts [hunk 2]
```diff
@@ -1,29 +1,13 @@
...
-import supabase from "@/supabaseClient";
 import { CalendarEvent } from "@/components/Calendar/Calendar";
+import { ParcelWithClientName } from "@/app/calendar/getParcelsWithCollectionDate";
 
-export type ParcelWithClientName = Schema["parcels"] & { clients: { full_name: string } | null };
-
-const COLLECTION_DURATION_MS = 30 * 60 * 1000;
-
-export interface LocationColorMap {
+interface LocationColorMap {
```
Comment (SamYap-Softwire) [inline src/app/calendar/parcelsToCollectionEvents.ts:4]: used in another file - add export

File: src/components/DataInput/inputHandlerFactories.ts [hunk 3]
```diff
@@ -1,64 +1,64 @@
...
 import { SelectChangeEvent } from "@mui/material";
 
-type valueSetter = {
+interface ValueSetter {
     (value: string): void;
-};
+}
 
-type booleanSetter = {
+interface BooleanSetter {
```
Comment (Var1377) [inline src/components/DataInput/inputHandlerFactories.ts:8]: Remove this and make ValueSetter generic:

```
type ValueSetter<T = string> = (data: T) => void;
```

File: src/components/Form/GenericFormCard.tsx [hunk 4]
```diff
@@ -1,11 +1,13 @@
...
+
 import React from "react";
 import { FormSubheading, FormText, StyledCard } from "@/components/Form/formStyling";
 import styled from "styled-components";
 
 interface Props {
     title: string;
     required: boolean;
-    children: React.ReactElement;
+    children: React.ReactNode;
```
Comment (Var1377) [inline src/components/Form/GenericFormCard.tsx:10]: children should usually be optional, not sure if that's what you want here

File: src/components/Title/Title.tsx [hunk 5]
```diff
@@ -13,7 +13,7 @@ const TitleHeader = styled.h1`
     font-size: min(6.5vw, 2rem);
 `;
 
-const Title: React.FC<TitleProps> = (props) => {
+const Title: React.FC<Props> = (props) => {
     return <TitleHeader>{props.children}</TitleHeader>;
 };
```
Comment (Var1377) [inline src/components/Title/Title.tsx:19]: Can we just `export default TitleHeader`

**Review by Var1377 — CHANGES_REQUESTED** [review]

File: src/app/clients/ActionBar.tsx [hunk 6]
```diff
@@ -162,7 +162,7 @@ const ActionBar: React.FC<Props> = ({ selected, data }) => {
             </Menu>
             {actionAnchorElement ? (
                 <Menu
-                    open={actionAnchorElement !== null}
+                    open
```
[the remark this answers is not in the document]
Comment (SamYap-Softwire) [reply src/app/clients/ActionBar.tsx:165]: it's not a modal :)

File: src/components/Form/GenericFormCard.tsx (hunk 4 above)
Comment (SamYap-Softwire) [reply src/components/Form/GenericFormCard.tsx:10]: yeah i think i want there to always be a child :)

File: src/app/calendar/getParcelsWithCollectionDate.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import supabase from "@/supabaseServer";
+import { Schema } from "@/database_utils";
+
+export type ParcelWithClientName = Schema["parcels"] & { clients: { full_name: string } | null };
```
[the remark this answers is not in the document]
Comment (SamYap-Softwire) [reply src/app/calendar/getParcelsWithCollectionDate.ts:4]: i think it's fine to use types for these cases

File: src/app/calendar/page.tsx [hunk 8]
```diff
@@ -1,7 +1,9 @@
 import { Metadata } from "next";
 import React from "react";
 import ParcelCalendar from "@/app/calendar/ParcelCalendar";
-import { getParcelsWithCollectionDate } from "@/app/calendar/parcelCalendarFunctions";
+import getParcelsWithCollectionDate from "@/app/calendar/getParcelsWithCollectionDate";
+
+export const dynamic = "force-dynamic";
```
[the remark this answers is not in the document]
Comment (SamYap-Softwire) [reply src/app/calendar/page.tsx:6]: I'll sort out the supabase things in a different PR :)

[the remark this answers is not in the document]
Comment (Var1377) [reply src/app/calendar/page.tsx:6]: Could you then get rid of "force-dynamic" here? (Unless it's totally broken without it)

File: src/components/Hooks/useForceRerender.ts [hunk 9]
```diff
@@ -1,3 +1,5 @@
+"use client";
+
 import { useCallback, useState } from "react";
 
 const useForceRerender = (): (() => void) => {
```
Comment (Var1377) [inline src/components/Hooks/useForceRerender.ts:5]: this is actually no longer used, you can get rid of it

**Review by Var1377 — APPROVED** [review]
Just a couple minor comments

File: src/app/calendar/page.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (SamYap-Softwire) [reply src/app/calendar/page.tsx:6]: unfortunately it is TT (can't even run build)

**Review by daniel-conlan-softwire — APPROVED** [review]

**Pull request closed without merging**